### PR TITLE
Staging Sites: Fix incorrect check for active plugins array

### PR DESCRIPTION
**Pull request by wojtekn**
Related to https://github.com/Automattic/dotcom-forge/issues/2003

## Proposed Changes

In this diff, I propose to fix a condition that checks for active Jetpack plugins. Currently, it doesn't check for array length, so it returns `true` even for an empty array.

## Testing Instructions

1. Create one or more staging sites using the wpcalypso environment
2. Open a `/sites` page 
3. In another browser tab, open any site and then click "Switch site"
4. Compare list of sites in sidebar and Sites page and ensure sites are displayed in the same order
5. Filter both lists by "staging-" and confirm that the same sites are displayed in both places

Screenshot 2023-03-20 at 14 19 18

## Pre-merge Checklist

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-ajp-p2)?
](https://github.com/Automattic/dotcom-forge/issues/2003)

### CODE REVIEW

**Review by kozer — APPROVED** [review]

File: client/my-sites/picker/picker.jsx [hunk 1]
```diff
@@ -92,7 +92,7 @@ class SitePicker extends Component {
 	filterSites = ( site ) => {
 		// Filter out the sites on WPCOM that don't have full Jetpack plugin installed
 		// Such sites should work fine on Jetpack Cloud
-		return getJetpackActivePlugins( site ) ? isJetpackSiteOrJetpackCloud( site ) : true;
+		return getJetpackActivePlugins( site )?.length ? isJetpackSiteOrJetpackCloud( site ) : true;
```
Comment (danielbachhuber) [inline client/my-sites/picker/picker.jsx]: Interesting!

Should we take this as an opportunity to abstract this filter function and write some tests around it?

Comment (wojtekn) [reply client/my-sites/picker/picker.jsx]: Thanks for pointing that out. I went a step further and refactored this code. `getJetpackActivePlugins` was used only to check for any active Jetpack plugins, so I renamed this selector to `hasJetpackActivePlugins`, fixed implementation, and added tests to check three cases.

**Review by danielbachhuber — APPROVED** [review]
Code looks good to me. I didn't test it, though.

**Review by sejas — CHANGES_REQUESTED** [review]
@wojtekn , 
~I think this PR introduces a bug in Jetpack cloud. It shows a wrong number of sites in Jetpack cloud site switcher.~

~Here are the screenshots.~


| **Pro** | **This PR** |
|---|---|
| Screenshot 2023-03-23 at 09 52 09 | Screenshot 2023-03-23 at 09 51 53 |


### ~Steps to reproduce~

~1. Create a Jetpack site~
~2. Go to https://cloud.jetpack.com/landing~
~3. Click on your site~
~4. Click on `Switch site`~
~5. Observe the `All my sites` displays a correct number.~
~6. Repeat the steps in Jetpack cloud live~
~7. Observe that it shows a higher number of sites.~

--

UPDATE: this bug happens in jetpack cloud live and not in localhost. It was not introduced by this PR.

**Review by sejas — APPROVED** [review]
Approved since the error on `All my sies` number is not introduced by this PR and is a bug on calypso live itself.
It works as expected in localhost.

Well done with this fix and catching this! @wojtekn

**Comment by wojtekn** [comment]
Thnks for testing and confirming this @sejas !

**Pull request merged**